cpp/include/algorithm/number_theory/fibonacci_golden.hpp

```diff
@@ -0,0 +1,73 @@
+/*
+    Fibonacci number
+    -----------------
+    Find the N-th Fibonacci number, given the value of N. The Fibonacci
+    sequence is characterized by the fact that every number after the
+    first two is the sum of the two preceding ones.
+
+    It uses the fact that the round off value of the next fibonacci number
+    is f_(n+1) = f_(n)*(phi) where phi is golden ratio
+
+
+    Time complexity
+    ---------------
+    O(log N), where N is the term of the Fibonacci sequence to calculate.
+
+    Space complexity
+    ----------------
+    O(1).
+*/
+
+#ifndef FIBONACCI_GOLDEN_HPP
+#define FIBONACCI_GOLDEN_HPP
+
+#include <cmath>
+typedef long long int ULL;
+
+const int MAX_N = 64; // fibonacci(64) cannot be calculated due to limitation of precision of computers for irrational number phi (GOLDEN RATIO)
+const long double phi = 1.618033988749894;
+
+long double ___modExp_____(const int ex) // Name is unconventional to prevent accidental clash with a function name in implementing class
```

Comment:
Please add a comment explaining what this function does and how it works. Also, I'd see if you can come up with a more descriptive identifier than `modExp`.